.github/workflows/test.yml

```diff
@@ -2,43 +2,41 @@ name: RSpec tests
 
 on:
   push:
-    branches: [ master ]
+    branches: [ master, ruby-31-support ]
   pull_request:
     branches: [ master ]
 
 jobs:
   ruby:
     runs-on: ubuntu-latest
-    timeout-minutes: 5
+    timeout-minutes: ${{ matrix.timeout }}
     strategy:
       fail-fast: false
       matrix:
        include:
-         - docker-image: 'silvioq/ruby-1.8.7'
-           bundler-version: '1.17.3'
-         - docker-image: 'library/ruby:1.9'
-           bundler-version: '1.17.3'
-         - docker-image: 'library/ruby:2.0'
-           bundler-version: '1.17.3'
-         - docker-image: 'library/ruby:2.2'
-           bundler-version: '1.17.3'
-         - docker-image: 'library/ruby:2.3'
-         - docker-image: 'library/ruby:2.4'
-         - docker-image: 'library/ruby:2.5'
-         - docker-image: 'library/ruby:2.6'
-         - docker-image: 'library/ruby:2.7'
-         - docker-image: 'library/ruby:3.0'
-         - docker-image: 'flavorjones/truffleruby:stable'
-         - docker-image: 'bitnami/jruby:latest'
-    container: ${{ matrix.docker-image }}
+         - ruby: 2.5
+           timeout: 5
+         - ruby: 2.6
+           timeout: 5
+         - ruby: 2.7
+           timeout: 5
+         - ruby: '3.0'
+           timeout: 5
+         - ruby: 3.1
+           timeout: 5
+         - ruby: jruby
+           timeout: 5
+         - ruby: truffleruby
+           timeout: 60
     steps:
     - uses: actions/checkout@v2
-    - name: Install bundler ${{ matrix.bundler-version }}
-      run: gem install bundler ${{ matrix.bundler-version && format('{0} {1}', '-v', matrix.bundler-version) || '' }}
-    - name: Install dependencies
-      run: bundle install
+    - uses: ruby/setup-ruby@v1.96.0
```

Comment:
Let's use `@v1` and remove truffleruby for now, I guess this was tried to address https://github.com/deivid-rodriguez/mail/runs/5230121239?check_suite_focus=true?

That failure is an unfortunate consequence of depending on default gems explicitly, and the existing TruffleRuby release can do nothing about this (truffleruby-head doesn't have the issue, it just always uses the stdlib version). Well, it'd be possible to modify the upstream gem but it's unclear what's best at this point (ignore the gem & use the stdlib, or use the gem C ext but that's slower for some cases, or integrate in some other way).

Reply:
There's something definitely wrong in `ruby/setup-ruby`. Bundle install is working perfect with bundler 2.3.7 on truffleruby. So I tried to use it by specifying `bundler: 2.3.7`, and `ruby/setup-ruby` ignored my explicit choice and still used an old Bundler. Version 1.96.0 does not have that bug.

Reply:
It's due to https://github.com/ruby/setup-ruby/releases/tag/v1.97.0 / https://github.com/oracle/truffleruby/issues/2586 / https://github.com/rake-compiler/rake-compiler/issues/202.
`rake-compiler` started to mutate RbConfig in a way RubyGems can't find where to insert gems anymore.
One fix is to `@gem_prelude_index` but it can't be done for an existing release.
Using the Bundler shipped with TruffleRuby avoids multiple Bundler version and the problem, so that's the workaround for 22.0.
The output does say `Using Bundler 2 shipped with truffleruby-22.0.0.2` but doesn't explain why, should probably link to the issue.

Reply:
Alright, but if I explicitly say the version of Bundler that I want because I'm having issues with the default version, and not hitting the issues it's preventing (because of not using `rake-compiler`, for example), shouldn't that be respected?

Reply:
Unsure, because e.g. Bundler 1 is enforced on old Rubies (no mater the bundler input) because Bundler 2 wouldn't work or would just lead to confusing errors. Similar here, although it is less clear because latest Bundler could potentially work, if no `rake-compiler` involved, but that's not easy to find out (before the `bundle lock`).
I hate the situation, but it's also crazy rake-compiler (and RubyGems too, the code is copied from RubyGems) mutate RbConfig and there seem to be limited workarounds for that besides not mutating anymore.

Reply:
I really appreciate trying to provide a good default experience by adding workarounds for external issues, but the current behavior feels way too much to me. In any case, I'm happy sticking to 1.96 here until the next release of truffleruby!